server/src/api/util/level.js

```diff
@@ -0,0 +1,69 @@
+const { LEVEL_EXP } = require('../constants/levels');
+
+function getLevel(experience) {
+  // `experience` needs to be defined as a `number`
+  if (!experience || typeof experience !== 'number') {
+    return 0;
+  }
+
+  let index;
+
+  for (index = 0; index < LEVEL_EXP.length; index += 1) {
+    if (LEVEL_EXP[index + 1] > experience) {
+      break;
+    }
+  }
+
+  return Math.min(index, 99);
+}
+
+function getCombatLevel(playerExperiences) {
```

Comment:
This still feels uncomfortably messy, maybe we can use [pick](https://underscorejs.org/#pick) to extract the fields we need and [transform](https://lodash.com/docs/4.17.15#transform) to transform the objects. Then we can extract the values when calculating. So something like
```
const combatExperience = _.pick(playerExperiences, attackExperience, ..., magicExperience);
const levels = _.transform(combatExperience, (r, v, k) => {
    r[k.replace("Experience", "Level")] = getLevel(v);
});
...
```